src/pyinit.jl

```diff
@@ -125,6 +126,17 @@ function Py_Finalize()
 end
 
 function __init__()
+    # Clear out global states.  This is required only for PyCall
+    # AOT-compiled into system image.
+    _finalized[] = false
+    empty!(_namespaces)
+    empty!(eventloops)
+    empty!(npy_api)
+    empty!(pycall_gc)
+    empty!(pyexc)
+    empty!(pytype_queries)
+    empty!(permanent_strings)
```

Comment:
Why is this required?

Reply:
The changes in https://github.com/JuliaPy/PyCall.jl/pull/651/commits/b0e131ac8c89b6247a436ae60f270e65991955a3 are the only ones required for making PyCall compilable with PackageCompiler.  The ones in https://github.com/JuliaPy/PyCall.jl/pull/651/commits/a9e9419a40192d3397d6d9f6b4c6a805af02b728 are not required but added for consistency and possible similar problems.  It seems that at least adding them does not break PyCall.